ResourceStatusSystem/Client/SiteStatus.py

```diff
@@ -0,0 +1,243 @@
+
+  Provides methods to easily interact with the RSS
+
+"""
+
+from DIRAC                                                  import gConfig, gLogger, S_OK, S_ERROR
+from DIRAC.Core.Utilities.DIRACSingleton                    import DIRACSingleton
+from DIRAC.Core.DISET.RPCClient                             import RPCClient
+from DIRAC.ConfigurationSystem.Client.Helpers.Operations    import Operations
+from DIRAC.ResourceStatusSystem.Client.ResourceStatusClient import ResourceStatusClient
+from DIRAC.ResourceStatusSystem.Client.ResourceStatus       import ResourceStatus
+from DIRAC.ResourceStatusSystem.Utilities.RssConfiguration  import RssConfiguration
+from DIRAC.Core.Utilities                                   import DErrno
+
+__RCSID__ = '$Id: $'
+
+class SiteStatus( object ):
+  """
+  RSS helper to interact with the 'Site' family on the DB. It provides the most
+  demanded functions and a cache to avoid hitting the server too often.
+
+  It provides four methods to interact with the site statuses:
+  * getSiteStatuses
+  * isUsableSite
+  * getUsableSites
+  * getSites
+  """
+
+  __metaclass__ = DIRACSingleton
+
+  def __init__( self ):
+    """
+    Constructor, initializes the rssClient.
+    """
+
+    self.log = gLogger.getSubLogger( self.__class__.__name__ )
+    self.rssConfig = RssConfiguration()
+    self.__opHelper = Operations()
+    self.rssFlag = ResourceStatus().rssFlag
+    self.rsClient = ResourceStatusClient()
+    self.wmsAdministrator = RPCClient( 'WorkloadManagement/WMSAdministrator' )
+
+  def getSiteStatuses( self, siteNamesList = None ):
+    """
+    Method that queries the database for status of the sites in a given list.
+    If the input is None, it is interpreted as * ( all ).
+
+    If match is positive, the output looks like:
+    {
+     'test1.test1.org': 'Active',
+     'test2.test2.org': 'Banned',
+    }
+
+    examples
+      >>> siteStatus.getSiteStatuses( [ 'test1.test1.uk', 'test2.test2.net', 'test3.test3.org' ] )
+          S_OK( { 'test1.test1.org': 'Active', 'test2.test2.net': 'Banned', 'test3.test3.org': 'Active' }  )
+      >>> siteStatus.getSiteStatuses( 'NotExists')
+          S_ERROR( ... ))
+      >>> siteStatus.getSiteStatuses( None )
+          S_OK( { 'test1.test1.org': 'Active',
+                  'test2.test2.net': 'Banned', },
+                  ...
+                }
+              )
+
+    :Parameters:
+      **siteNamesList** - `list`
+        name(s) of the sites to be matched
+
+    :return: S_OK() || S_ERROR()
+    """
+
+    if not siteNamesList:
+
+      if self.rssFlag:
+        siteStatusDict = self.rsClient.selectStatusElement( 'Site', 'Status', meta = { 'columns' : [ 'Name', 'Status' ] } )
+      else:
+        siteStatusDict = self.wmsAdministrator.getSiteMask()
+
+      if not siteStatusDict['OK']:
+       return S_ERROR(DErrno.ERESGEN, 'selectStatusElement failed')
+      else:
+       siteStatusDict = siteStatusDict['Value']
+
+      return S_OK( dict(siteStatusDict) )
```

Comment:
In case that `self.rssFlas` is `False`, then:
```
In [11]: siteStatusDict = wmsAdministrator.getSiteMask()['Value']

In [12]: type(siteStatusDict)
Out[12]: list

In [13]: dict(siteStatusDict)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-13-32df9a61da44> in <module>()
----> 1 dict(siteStatusDict)

ValueError: dictionary update sequence element #0 has length 15; 2 is required
```
So this can't work!